code/datums/abilities/flock/flockmind.dm

```diff
@@ -69,76 +123,99 @@
 		boutput(holder.owner, "<span class='alert'>The flock can't convert this.</span>")
-		return 1
+		return TRUE
 	if(isfeathertile(T))
 		boutput(holder.owner, "<span class='alert'>This tile has already been converted.</span>")
-		return 1
-	if(F)
-		var/datum/flock/flock = F.flock
-		if(flock)
-			flock.togglePriorityTurf(T)
+		return TRUE
+	if (!(T in F.flock.priority_tiles))
+		for (var/name in F.flock.busy_tiles)
+			if (T == F.flock.busy_tiles[name])
+				boutput(holder.owner, "<span class='alert'>This tile is already scheduled for conversion!</span>")
+				return TRUE
+	F.flock?.togglePriorityTurf(T)
 
 /////////////////////////////////////////
 
 /datum/targetable/flockmindAbility/designateEnemy
 	name = "Designate Enemy"
-	desc = "Mark someone as an enemy."
+	desc = "Mark or unmark someone as an enemy."
 	icon_state = "designate_enemy"
 	cooldown = 0
-	//sticky = 1
 
 /datum/targetable/flockmindAbility/designateEnemy/cast(atom/target)
 	if(..())
-		return 1
-	var/mob/living/M = target
-	var/mob/living/intangible/flock/flockmind/F = holder.owner
-	if(isliving(M))
-		if(F)
-			var/datum/flock/flock = F.flock
-			if(flock)
-				flock.updateEnemy(M)
-	else
-		boutput(holder.owner, "<span class='alert'>That isn't a valid target.</span>")
-		return 1
+		return TRUE
+
+	var/M = target
+	var/mob/living/intangible/flock/F = holder.owner
+
+	if (!(isliving(M) || iscritter(M) || isvehicle(M)) || isflock(M) || isintangible(M))
+		boutput(F, "<span class='alert'>That isn't a valid target.</span>")
+		return TRUE
+
+	var/datum/flock/flock = F.flock
+
+	if (!flock)
+		return TRUE
+
+	if (flock.isEnemy(M))
+		flock.removeEnemy(M)
+		return
+
+	flock.updateEnemy(M)
 
 /////////////////////////////////////////
 
 /datum/targetable/flockmindAbility/partitionMind
 	name = "Partition Mind"
 	desc = "Divide and conquer."
 	icon_state = "awaken_drone"
-	targeted = 0
+	cooldown = 60 SECONDS
+	targeted = FALSE
+	///Are we still waiting for ghosts to respond
+	var/waiting = FALSE
 
 /datum/targetable/flockmindAbility/partitionMind/cast(atom/target)
-	if(..())
-		return 1
+	if(waiting || ..())
+		return TRUE
+
+	if(!holder.pointCheck(100))
+		return TRUE
+
 	var/mob/living/intangible/flock/flockmind/F = holder.owner
-	if(F)
+	waiting = TRUE
+	SPAWN(0)
 		F.partition()
-	else
-		return 1 // not sure what happened here
+		waiting = FALSE
 
 /////////////////////////////////////////
 
 /datum/targetable/flockmindAbility/healDrone
 	name = "Concentrated Repair Burst"
 	desc = "Fully heal a drone through acceleration of its repair processes."
 	icon_state = "heal_drone"
+	cooldown = 20 SECONDS
 
 /datum/targetable/flockmindAbility/healDrone/cast(mob/living/critter/flock/drone/target)
 	if(..())
-		return 1
+		return TRUE
 	if(!istype(target))
-		return 1
+		return TRUE
+	if (target.get_health_percentage() >= 1)
+		boutput(holder.owner, "<span class='notice'>[target.real_name] has no damage!</span>")
+		return TRUE
+	if (isdead(target))
```

Comment:
Should add a message for the user here, I can imagine "Why can't I heal this dead drone" to be a reasonable mistake

Reply:
https://github.com/stonepillars/goon-flock/pull/317